doc/contributing.rst

```diff
@@ -1,221 +1,808 @@
+
+This will build the documentation *except* it will format (but not execute) the
+tutorial and example files. If you have created or modified an example or
+tutorial, you should instead run
+:samp:`PATTERN={<REGEX_TO_SELECT_MY_TUTORIAL>} make html_dev-pattern` to render
+all the documentation and additionally execute just your example or tutorial
+(so you can make sure it runs successfully and generates the output / figures
+you expect).
+
+After either of these commands completes, ``make show`` will open the
+locally-rendered documentation site in your browser. Additional ``make``
+recipes are available; run ``make help`` from the :file:`doc` directory or
+consult the `Sphinx-Gallery`_ documentation for additional details.
+
+
+.. _`github-workflow`:
+
+GitHub workflow
+^^^^^^^^^^^^^^^
+
+Nearly everyone in the community of MNE-Python contributors and maintainers is
+a working scientist, engineer, or student who contributes to MNE-Python in
+their spare time. For that reason, a set of best practices have been adopted to
+streamline the collaboration and review process. Most of these practices are
+common to many open-source software projects, so learning to follow them while
+working on MNE-Python will bear fruit when you contribute to other projects
+down the road. Here are the guidelines:
+
+- Search the `MNE-Python issues page`_ (both open and closed issues) in case
+  someone else has already started work on the same bugfix or feature. If you
+  don't find anything, `open a new issue`_ to discuss changes with maintainers
+  before starting work on your proposed changes.
+
+- Implement only one new feature or bugfix per pull request (PR). Occasionally
+  it may make sense to fix a few related bugs at once, but this makes PRs
+  harder to review and test, so check with MNE-Python maintainers first before
+  doing this. Avoid purely cosmetic changes to the code; they make PRs harder
+  to review.
+
+- It is usually better to make PRs *from* branches other than your master
+  branch, so that you can use your master branch to easily get back to a
+  working state of the code if needed (e.g., if you're working on multiple
+  changes at once, or need to pull in recent changes from someone else to get
+  your new feature to work properly).
+
+- In most cases you should make PRs *into* the upstream's master branch, unless
+  you are specifically asked by a maintainer to PR into another branch (e.g.,
+  for backports or maintenance bugfixes to the current stable version).
+
+- Don't forget to include in your PR a brief description of the change in the
+  :doc:`changelog <whats_new>` (:file:`doc/whats_new.rst`).
+
+- Our community uses the following commit tags and conventions:
+
+  - Work-in-progress PRs should be created as `draft PRs`_ and the PR title
+    should begin with ``WIP``
+
+  - When you believe a PR is ready to be reviewed and merged, `convert it
+    from a draft PR to a normal PR`_, and change its title to begin with
+    ``MRG``
```

Comment:
and make a comment on GitHub asking for reviews, as maintainers are not notified of title changes